### PR TITLE
add Clojure's as-> macro

### PR DESCRIPTION
**Pull request by gilch**
see #1047, and the docstring.

### CODE REVIEW

**Comment by gilch** [comment]
I suppose I should add tests.

**Comment by tuturto** [comment]
Looks nifty, but docs and some tests would be nice as I'm not 100% sure I understand all the implications.

**Comment by algernon** [comment]
A test would be very neat, but I'd :+1: it anyway.

File: hy/core/macros.hy [hunk 1]
```diff
@@ -30,6 +30,16 @@
         [hy.models.symbol [HySymbol]]
         [hy._compat [PY33 PY34]])
 
+(defmacro as-> [head name &rest rest]
+  "Expands to sequence of assignments to the provided name, starting with head.
+  The previous result is thus available in the subsequent form. Returns the
+  final result, and leaves the name bound to it in the local scope. This behaves
+  much like the other threading macros, but requires you to specify the threading
+  point per form via the name instead of always the first or last arument."
```
Comment (charness) [inline hy/core/macros.hy]: (Docstring typo: missing the _g_ in _argument_.)

**Pull request merged**